### PR TITLE
ROCANA-3468: Make Impala release build work

### PR DESCRIPTION
**Pull request by bradcupit**
We had a compile error in the Impala C++ code in Release mode. The same issue did not happen in Debug mode.

It turns out Cloudera already fixed the issue, so I cherry-picked their commit from the `cdh5-trunk` branch (their version of `master`).

To test I compiled the following ways and all worked:
- `bin/make_release.sh -clean`
- `bin/make_release.sh -clean -build_shared_libs`
- `bin/make_debug.sh` <br>
  Note: this one implicitly includes `-build_shared_libs`

### CODE REVIEW

**Comment by joey** [comment]
Was it a clean cherry-pick?

It'd be nice to reference the upstream JIRA in the commit message.

+1

**Comment by bradcupit** [comment]
It was a clean cherry pick

File: be/src/exprs/in-predicate-benchmark.cc [hunk 1]
```diff
@@ -158,6 +158,7 @@
 #include <gutil/strings/substitute.h>
 
 #include "exprs/in-predicate.h"
+#include "exprs/in-predicate-ir.cc"
```
Comment (bradcupit) [inline be/src/exprs/in-predicate-benchmark.cc:161]: This is the actual fix for the compile error in release mode. It looks to me like `in-predicate-ir.cc` is in the Makefiles generated for Release builds, so I'm not sure why the linker couldn't find it and why we need this line.

See [ROCANA-3468](https://scalingdata.atlassian.net/browse/ROCANA-3468) for the compile error.

**Comment by bradcupit** [comment]
Added the JIRA ticket name to the commit

**Pull request merged**